code/game/gamemodes/objective.dm

```diff
@@ -164,6 +164,32 @@ GLOBAL_LIST_INIT(potential_theft_objectives, (subtypesof(/datum/theft_objective)
 		return FALSE
 	return TRUE
 
+/datum/objective/assassinate/assassinateonce
+	name = "AssassinateOnce"
+	martyr_compatible = 1
+	var/won = FALSE
+
+/datum/objective/assassinate/assassinateonce/find_target(list/target_blacklist)
```

Comment:
```suggestion
/datum/objective/assassinateonce/find_target(list/target_blacklist)
```
Otherwise the logic of `assasinate` will be executed first (because the first statement in your proc is ..())

Reply:
Also change your new objective path everywhere.

Reply:
do you think I need the full completion check?